tests/test_commands.py

```diff
@@ -0,0 +1,188 @@
+import os
+import signal
+import shutil
+import subprocess
+import sys
+import time
+
+import pytest
+import requests
+import sh
+
+
+FIXTURES = 'tests/fixtures'
+WEBSITE = os.path.join(FIXTURES, 'website.py')
+BUILDDIR = os.path.join(FIXTURES, '_build')
+INDEX = os.path.join(BUILDDIR, 'index.html')
+CNAME = os.path.join(BUILDDIR, 'CNAME')
+
+
+def parametrized_fixture(*args, **kwargs):
+    @pytest.fixture(scope='module', params=args + tuple(kwargs.keys()))
+    def fixture(request):
+        if request.param in kwargs:
+            return kwargs[request.param]
+        return request.param
+    return fixture
+
+
+class Cname(str):
+    '''
+    Helper class that is str, but has different truthfulness
+    '''
+    def __bool__(self):
+        return self == ' --cname'
+
+
+cname = parametrized_fixture(cname=Cname(' --cname'),
+                             no_cname=Cname(' --no-cname'))
+serve_command = parametrized_fixture('serve', freeze_serve='freeze --serve')
+port = parametrized_fixture(8001, 8080)
+domain = parametrized_fixture('foo.bar', 'spam.eggs')
+protocol = parametrized_fixture('http', 'https')
+
+
+class ElsaRunner:
+    '''
+    Class for elsa fixture enabling multiple runs and exit status check
+    '''
+    def __init__(self):
+        self.runs = []
+
+    def run(self, command):
+        print('COMMAND: python website.py', command)
+        proc = subprocess.Popen([sys.executable, WEBSITE] + command.split())
```

Comment:
Have you considered making `command` a list, instead of a string? Lists generally work better when dealing with `subprocess` args. There's a bit more typing in `['elsa', 'freeze', 'serve']`, but not having to do `split()` or remember to keep a spaces at the right end of a string makes things much more maintainable, IMO.